src/shared/Microsoft.Git.CredentialManager/Interop/Windows/WindowsSystemPrompts.cs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Git.CredentialManager.Interop.Windows.Native;
+
+namespace Microsoft.Git.CredentialManager.Interop.Windows
+{
+    public class WindowsSystemPrompts : ISystemPrompts
+    {
+        public bool ShowCredentialPrompt(string resource, string userName, out ICredential credential)
+        {
+            EnsureArgument.NotNullOrWhiteSpace(resource, nameof(resource));
+
+            string message = $"Enter credentials for '{resource}'";
+
+            var credUiInfo = new CredUi.CredentialUiInfo
+            {
+                BannerArt = IntPtr.Zero,
+                CaptionText = "Git Credential Manager", // TODO: make this a parameter?
+                Parent = IntPtr.Zero, // TODO: get the parent window handle
```

Comment:
We'll want to parent this correctly so that Windows can always position the dialog above the parent HWND.